pkg/scheduler/internal/cache/cache.go

```diff
@@ -537,9 +537,7 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 
 	currState, ok := cache.podStates[key]
 	switch {
-	// An assumed pod won't have Delete/Remove event. It needs to have Add event
-	// before Remove event, in which case the state would change from Assumed to Added.
-	case ok && !cache.assumedPods.Has(key):
+	case ok:
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
 			klog.Errorf("Pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
 			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
```

Comment:
this klog.Fatalf here is worrying... is currState or pod from an informer cache?

Reply:
`currState.pod` is the internal cache and `pod` is from the event (so from the informer in the case of `DeletedFinalStateUnknown`)

If the pod was assumed, `currState.pod` would have a name and `pod` might not, if events were missed. I think we can skip the Fatal in that case.

Reply:
we can skip the fatal here.

Reply:
Fine with skip fatal.

Reply:
Done